http/client.go

```diff
@@ -290,7 +292,13 @@ func (c *Client) sendRateLimitedRequest(
 		if err != nil {
 			return nil, errors.Newf(err, "failed executing the request %s", URL)
 		}
-		if resp.StatusCode != http.StatusRequestEntityTooLarge || resp.Header.Get("Retry-After") == "" {
+
+		retryAfterStatusCodes := set.NewInts(
```

Comment:
this would be a bit more efficient as a switch:

```go
switch resp.StatusCode {
case http.StatusRequestEntityTooLarge,
         http.StatusForbidden,
         http.StatusServiceUnavailable,
         http.StatusTooManyRequests:
    if resp.Header.Get("Retry-After") == "" {
        return resp, nil
    }
default:
    return resp, nil
}
```